src/coreclr/jit/codegencommon.cpp

```diff
@@ -1956,6 +1963,116 @@ void CodeGen::genGenerateMachineCode()
     /* The code is now complete and final; it should not change after this. */
 }
 
+//----------------------------------------------------------------------
+// genUpdateLiveRangesForTruncatedIGs -- find any instruction groups that have
+//   had their count altered and updated live ranges that refer to those groups
+//
+void CodeGen::genUpdateLiveRangesForTruncatedIGs()
+{
+#ifdef DEBUG
+    if (compiler->verbose)
+    {
+        printf("*************** In genUpdateLiveRangesForTruncatedIGs()\n");
+    }
+#endif
+    bool updatedGroup = false;
+
+    if (compiler->lvaCount > 0)
+    {
+        insGroup* ig = GetEmitter()->emitIGlist;
+        while (ig != nullptr)
+        {
+            if (ig->igFlags & IGF_UPD_ICOUNT)
```

Comment:
Might be nice to reduce nesting by making this:

```suggestion
            if ((ig->igFlags & IGF_UPD_ICOUNT) == 0)
            {
              continue;
            }
```

Reply:
Done.